File: app/src/main/java/ys/moire/common/di/ApplicationComponent.java
package ys.moire.common.di;

import dagger.Component;
import ys.moire.presentation.ui.main.MainActivity;
import ys.moire.presentation.ui.preferences.PreferencesActivity;

/**
 * dagger application component.
 */
@Component (modules = {ApplicationModule.class})
public interface ApplicationComponent {

    void inject(MainActivity mainActivity);

    void inject(PreferencesActivity preferencesActivity);
}

File: app/src/main/java/ys/moire/common/di/ApplicationComponent.kt
package ys.moire.common.di

import dagger.Component
import ys.moire.presentation.ui.main.MainActivity
import ys.moire.presentation.ui.preferences.PreferencesActivity

/**
 * dagger application component.
 */
@Component(modules = arrayOf(ApplicationModule::class))
interface ApplicationComponent {

    fun inject(mainActivity: MainActivity)

    fun inject(preferencesActivity: PreferencesActivity)
}
